pkg/timestream/datasource.go

```diff
@@ -60,7 +61,23 @@ func NewServerInstance(s backend.DataSourceInstanceSettings) (instancemgmt.Insta
 		Runner: &timestreamRunner{
 			querySvc: func(region string) (client *timestreamquery.TimestreamQuery, err error) {
 
-				sess, err := sessions.GetSession(region, settings.AWSDatasourceSettings)
+				httpClientProvider := sdkhttpclient.NewProvider()
+				httpClientOptions, err := settings.Config.HTTPClientOptions()
+				if err != nil {
+					backend.Logger.Error("failed to create HTTP client options", "error", err.Error())
+					return nil, err
+				}
+				httpClient, err := httpClientProvider.New(httpClientOptions)
+				if err != nil {
+					backend.Logger.Error("failed to create HTTP client", "error", err.Error())
+					return nil, err
+				}
+
+				sess, err := sessions.GetSession(awsds.SessionConfig{
+					Settings:      settings.AWSDatasourceSettings,
+					HTTPClient:    httpClient,
+					UserAgentName: aws.String("Timestream"),
```

Comment:
I believe the previous user agent was `GrafanaTimestream/vx.x.x`. Is the `Grafana` prefix added by the aws-sdk?

Reply:
mmh, I forgot to actually remove  the other user agent. Before it was "GrafanaTimestream/1.0.0", now it will be "aws-sdk-go/$aws-sdk-version ($go-version; $OS;) TimeStream/$timestream-version-$git-hash Grafana/$grafana-version"

As you can see, there is much more info now (and it's what we are setting for the other plugins). This should be fine as long as we announce the change?

Reply:
Looks like a nice improvement. Yeah I think it should be fine